x-pack/test/functional/page_objects/infra_logs_page.ts

```diff
@@ -18,8 +18,21 @@ export function InfraLogsPageProvider({ getPageObjects, getService }: FtrProvide
       await pageObjects.common.navigateToApp('infraLogs');
     },
 
-    async navigateToTab(logsUiTab: LogsUiTab) {
-      await pageObjects.common.navigateToUrlWithBrowserHistory('infraLogs', `/${logsUiTab}`);
+    async navigateToTab(logsUiTab: LogsUiTab, params?: Record<string, RisonValue>) {
+      let queryString = '';
```

Comment:
Do we want to use `querystring.stringify` for this?

Reply:
Remember that we need to encode the values as `rison`. Otherwise they are not read.

With that in mind, the versions with and without `querystring` are not that different.

```ts
// Without (current)
let queryString = '';
if (params) {
  queryString = Object.keys(params).reduce((qs, key, idx) => {
    // Encode string and params as we go
    qs += (idx > 0 ? '&' : '') + `${key}=${encode(params[key])}`;
    return qs;
  }, '?');
}

// With (new)
let queryString = '';
if (params) {
  const encodedParams = Object.keys(params).reduce((p, key) => {
	// Encode params first
    p[key] = encode(params[key]);
    return p;
  }, {});
  
  // Encode string later
  queryString = '?' + querystring.stringify(encodedParams);
}
```

I lean more to the first approach, but if you think it's less obvious I don't mind changing it.

----

Note: the `querystring.stringify()` accepts [a function as an argument to encode the values](https://nodejs.org/api/querystring.html#querystring_querystring_stringify_obj_sep_eq_options). I tried doing that and passing `rison-node#encode`, but it still doesn't work, because according to the documentation:

> Any other input values will be coerced to empty strings.

```
% node
> const querystring = require('querystring')
undefined
> const { encode } = require('rison-node')
undefined
> const params = { logPosition: { start: 'foo', end: 'bar'} }
undefined
> params
{ logPosition: { start: 'foo', end: 'bar' } }
> querystring.stringify(params, null, null, { encodeURIComponent: encode })
'logPosition=\'\''
>
```

Reply:
That's indeed not much shorter, but it has the advantage of being more discoverable. As we're seeing with recent changes like #58943, there are situation were we want to globally adjust the way URLs are manipulated. Using `querystring` would allow a search to also discover this location.

Reply:
I agree with your point. I'll change it 👍